### DISCUSSION TITLE
Need some help with binding events on MacOS with CustomTkinter

### DISCUSSION THREAD

**Discussion by Gasiro** [TomSchimansky/CustomTkinter#2477] [Q&A] [answered]
Hey guys,
I tried to design an note app and trying to write a function that would prevent my app from closing, so that I can pop up a window for user to save their unsaved notes. So it should pop up when I press <command+w> / <command+q>. However, the thing is that I can not capture any of these two keys. As I showed below, only <command+a> was detected and printed successfully. Neither   <command+w> / <command+q> works. Does anyone know how to do this?
<img width="633" alt="Screenshot 2024-06-18 at 18 04 01" src="https://github.com/TomSchimansky/CustomTkinter/assets/117976338/e5e6f6bb-1c9f-42c2-bd97-299706839015">

**Comment by Anonymous6598** [comment]
try replacing `"<Command-a>"` with `"<Super_L>"`. The same with `"Command-w"` but use `"<Super_R>"`. If it doesn't work, just write this:
```python
root1.bind("<Command_W>", on_closing)
root1.bind("<Command_A>", cmd_a)
``` 
Hope it worked.

**Reply by Anonymous6598** [reply]
Full code:
```python
root1 = ctk.CTk()

def on_closing(event=None):
      print("Command-w pressed")
def cmd_a(event=None):
      print("Command-a preesed")

root1.bind("<Command_W>", on_closing)  # root1.bind("<Super_L>", on_closing)  only if doesn't work
root1.bind("<Command_A>", cmd_a)          # root1.bind("<Super _R>", cmd_a)  only if doesn't work

```

**Reply by Gasiro** [reply]
It seems not working either. All other keys would be recognised, only except the two keys <command_w> <command_q> that usually close the window and app.

**Reply by Gasiro** [reply, marked as answer]
It seems not work either, but I found solution here, by using tkinter menubar to actually bind the command-w successfully:
https://stackoverflow.com/questions/71606975/tkinter-bind-mac-os-commandq
